### PR TITLE
ENG-1657: Clean up Linter Warnings

### PR DESCRIPTION
**Pull request by agiron123**
## Describe your changes and why you are making these changes
This PR cleans up frontend linter warnings.

## Related issue number (if any)
ENG-1657

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/Search/index.tsx [hunk 1]
```diff
@@ -86,7 +82,7 @@ export const filteredList = (
   matchOn: (item: any) => string,
   listItems: (item: any, idx: number) => JSX.Element,
   noItemsMessage: JSX.Element
-) => {
+): JSX.Element => {
```
Comment (vsreekanti) [inline src/ui/common/src/components/Search/index.tsx:85]: Out of curiosity what is `JSX.Element` vs `React.Element`?

File: src/ui/common/src/components/notifications/NotificationListItem.tsx [hunk 2]
```diff
@@ -39,39 +34,40 @@ export const NotificationListItem: React.FC<Props> = ({
...
-              {notification.workflowMetadata.name}
-            </Typography>
-          </Typography>
-        </Box>
-      ) : (
-        <Box />
-      );
-      break;
-    }
+  // NOTE: Looks like title (and all the code in this switch statement) was unused and was giving us a linter warning :)
```
Comment (vsreekanti) [inline src/ui/common/src/components/notifications/NotificationListItem.tsx:37]: If it's unused, can we just delete?

**Review by vsreekanti — APPROVED** [review]
👍 One question, one small thing inline.

File: src/ui/common/src/components/Search/index.tsx (hunk 1 above)
Comment (agiron123) [reply src/ui/common/src/components/Search/index.tsx:85]: It's a more generic version of React.Element.

More info, and probably more than you'd like to know about these two here:
https://stackoverflow.com/questions/58123398/when-to-use-jsx-element-vs-reactnode-vs-reactelement

As for which one to use? I'm not 100% sure, they both seem to do the job and I don't really have strong opinions here.

File: src/ui/common/src/components/notifications/NotificationListItem.tsx (hunk 2 above)
Comment (agiron123) [reply src/ui/common/src/components/notifications/NotificationListItem.tsx:37]: Sure can. Good catch.

**Pull request merged**